lib/codecs/src/encoding/format/cef.rs

```diff
@@ -0,0 +1,538 @@
+use crate::encoding::BuildError;
+use bytes::BytesMut;
+use chrono::SecondsFormat;
+use lookup::lookup_v2::ConfigTargetPath;
+use snafu::Snafu;
+use std::num::ParseIntError;
+use std::{collections::HashMap, fmt::Write};
+use tokio_util::codec::Encoder;
+use vector_core::{
+    config::DataType,
+    event::{Event, LogEvent, Value},
+    schema,
+};
+
+const DEFAULT_DEVICE_VENDOR: &str = "Datadog";
+const DEFAULT_DEVICE_PRODUCT: &str = "Vector";
+const DEFAULT_DEVICE_VERSION: &str = "0"; // Major version. TODO: find a way to get the actual vector version.
```

Comment:
Can we give a bit more context here? Not sure I understand this TODO.

Reply:
Also added more info to this TODO